### PR TITLE
User location

### PR DESCRIPTION
**Pull request by kzzztkt**

### CODE REVIEW

File: utils/getUserLocation.js
Comment (wallace-y) [inline utils/getUserLocation.js:1]: Got a TypeError: Cannot read property "isDevice" of undefined
Possible unhandled promise rejection

File: utils/getUserLocation.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Platform, Text, View, StyleSheet } from 'react-native';
+import Device from 'expo-device';
+import * as Location from 'expo-location';
+
+
+async function getUserLocation () {
+    console.log('in function');
+      if (Platform.OS === 'android' && !Device.isDevice) {
+        setErrorMsg(
```
Comment (wallace-y) [inline utils/getUserLocation.js:9]: where is this function defined? I can't see where it would be used

File: utils/getUserLocation.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      }
+      let { status } = await Location.requestForegroundPermissionsAsync();
+      if (status !== 'granted') {
+        setErrorMsg('Permission to access location was denied');
+        return;
+      }
+try {
+    const dynamicLocation = await Location.getCurrentPositionAsync({})
+    return [dynamicLocation.coords.latitude, dynamicLocation.coords.longitude];
+} catch {
```
Comment (wallace-y) [inline utils/getUserLocation.js:22]: I don#t think the error is being caught here which might explain the unhandled promise

**Review by wallace-y — CHANGES_REQUESTED** [review]
There are a few things and I don't think there Android specific.
- the first catch block is missing the (err) after the catch
- the setErrorMsg isn't defined anywhere - might be worth changing to an alertt?
- there is an unhandled promise rejection but the above might fix

File: utils/getUserLocation.js (hunk 1 above)
Comment (kzzztkt) [reply utils/getUserLocation.js:9]: Thats the declaration. Its called in the Home.jsx

File: utils/getUserLocation.js
Comment (kzzztkt) [reply utils/getUserLocation.js:1]: I should have mentioned @wallace-y you need to npm i if you didn't do that? Or did you?

File: utils/getUserLocation.js (hunk 1 above)
Comment (wallace-y) [reply utils/getUserLocation.js:9]: the setErrorMsg isn't defined anywhere - sorry should have been more specific

Comment (wallace-y) [reply utils/getUserLocation.js:9]: If I remove the isDevice error, and fix the catch err, then it flags that setErrorMsg is undefined too

Comment (wallace-y) [reply utils/getUserLocation.js:9]: Just looking in the docs and saw this which I think fixes the device error: import * as Device from 'expo-device';

**Review by wallace-y — APPROVED** [review]
Great work on this mate - really cool!

**Pull request merged**